src/crypto/crypto_keys.cc

```diff
@@ -1134,6 +1136,51 @@ void KeyObjectHandle::InitEDRaw(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(true);
 }
 
+void KeyObjectHandle::Equals(const FunctionCallbackInfo<Value>& args) {
+  std::shared_ptr<KeyObjectData> key =
+    Unwrap<KeyObjectHandle>(args.Holder())->Data();
+  std::shared_ptr<KeyObjectData> key2 =
+    Unwrap<KeyObjectHandle>(args[0].As<Object>())->Data();
+
+  KeyType keyType = key->GetKeyType();
+  CHECK_EQ(keyType, key2->GetKeyType());
+
+  bool ret;
+  switch (keyType) {
+    case kKeyTypeSecret: {
+      size_t size = key->GetSymmetricKeySize();
+      if (size == key2->GetSymmetricKeySize())
+        ret = CRYPTO_memcmp(
+          key->GetSymmetricKey(),
+          key2->GetSymmetricKey(),
+          size) == 0;
+      else
+        ret = false;
+      break;
+    }
+    case kKeyTypePublic:
+      // Fall through
+    case kKeyTypePrivate: {
+      EVP_PKEY* pkey = key->GetAsymmetricKey().get();
+      EVP_PKEY* pkey2 = key2->GetAsymmetricKey().get();
+      if (EVP_PKEY_id(pkey) == EVP_PKEY_id(pkey2)) {
+#if OPENSSL_VERSION_MAJOR >= 3
+          ret = EVP_PKEY_eq(pkey, pkey2) == 1;
```

Comment:
This does not necessarily return `0` or `1` as far as I can tell. Negative codes can indicate an unsupported operation.

Reply:
- `-1` if the key types are different, regardless of whether "key types" means public/private or the EVP_PKEY_id, we never run those use cases
- `-2` if the operation is not supported, would you suggest to throw this case instead?